nav2_system_tests/src/system/test_system_launch.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+
+from launch import LaunchDescription
+from launch import LaunchService
+import launch.actions
+import launch_ros.actions
+from launch_testing import LaunchTestService
+
+
+def generate_launch_description():
+    map_yaml_file = os.getenv('TEST_MAP')
+    world = os.getenv('TEST_WORLD')
+    use_sim_time = True
+
+    # Launch gazebo server for simulation
+    return LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so', world],
+            output='screen'),
+
+        # Launch navigation2 nodes
+        launch_ros.actions.Node(
+            package='tf2_ros',
+            node_executable='static_transform_publisher',
+            output='screen',
+            arguments=['0', '0', '0', '0', '0', '0', 'base_footprint', 'base_link']),
+
+        launch_ros.actions.Node(
+            package='tf2_ros',
+            node_executable='static_transform_publisher',
+            output='screen',
+            arguments=['0', '0', '0', '0', '0', '0', 'base_link', 'base_scan']),
+
+        launch_ros.actions.Node(
+            package='nav2_map_server',
+            node_executable='map_server',
+            node_name='map_server',
+            output='screen',
+            parameters=[{'use_sim_time': use_sim_time}, {'yaml_filename': map_yaml_file}]),
+
+        launch_ros.actions.Node(
+            package='nav2_amcl',
+            node_executable='amcl',
+            node_name='amcl',
+            output='screen',
+            parameters=[{'use_sim_time': use_sim_time}]),
+
+        launch_ros.actions.Node(
+            package='nav2_world_model',
+            node_executable='world_model',
+            node_name='world_model',
```

Comment:
For world model and dwb controller, you'll need to set parameters using a yaml file now, and you'll want to remove the `node_name` line

Reply:
Fixed